aten/src/ATen/native/cuda/BatchLinearAlgebra.cpp

```diff
@@ -2972,9 +2974,36 @@ static void lu_solve_looped_magma(const Tensor& b, const Tensor& lu, const Tenso
   });
 }
 
+namespace {
+
+c10::MaybeOwned<Tensor> maybe_expand_lu(const Tensor& b, const Tensor& lu) {
+  if (batchCount(b) != batchCount(lu)) {
```

Comment:
This works whenever `b` is a matrix, but not when `B` is a batch of vectors, right?

Reply:
When we reach this level of code `B` is always a matrix or batch of matrices.